app/scripts/views/settings.js

```diff
@@ -3,78 +3,223 @@
+  'views/decorators/allow_only_one_submit',
   'stache!templates/settings'
 ],
-function (Cocktail, Session, FormView, BaseView, AvatarMixin,
-  SettingsMixin, Template) {
+function ($, modal, Cocktail, p, Session, BaseView, AvatarMixin,
+  AvatarView, AvatarChangeView, AvatarCropView, AvatarCameraView, GravatarView,
+  GravatarPermissionsView, CommunicationPreferencesView, ChangePasswordView,
+  DisplayNameView, DeleteAccountView, SettingsMixin, allowOnlyOneSubmit,
+  Template) {
   'use strict';
 
   var t = BaseView.t;
 
-  var View = FormView.extend({
+  var PANEL_VIEWS = [
+    AvatarView,
+    DisplayNameView,
+    CommunicationPreferencesView,
+    ChangePasswordView,
+    DeleteAccountView,
+    AvatarChangeView,
+    AvatarCropView,
+    AvatarCameraView,
+    GravatarView,
+    GravatarPermissionsView
+  ];
+
+  function renderView(view) {
+    return view.render()
+      .then(function (shown) {
+        if (! shown) {
+          view.destroy(true);
+          return;
+        }
+        view.afterVisible();
+
+        return view;
+      });
+  }
+
+  var View = BaseView.extend({
     template: Template,
     className: 'settings',
 
+    FADE_OUT_SETTINGS: 250,
+
     initialize: function (options) {
       options = options || {};
 
       this._able = options.able;
+      this._subViewToShow = options.subView;
+      this._notifications = options.notifications;
+      this._panelViews = options.panelViews || PANEL_VIEWS;
+
+      this.on('navigate-from-subview', this._onNavigateFromSubview.bind(this));
+
+      if (this._notifications) {
+        this._notifications.on(this._notifications.EVENTS.PROFILE_CHANGE,
+          this._onProfileChange.bind(this));
+      }
     },
 
     context: function () {
       var account = this.getSignedInAccount();
-      var email = account.get('email');
 
       return {
-        email: email,
+        username: account.get('email'),
         showSignOut: ! account.isFromSync(),
         communicationPrefsVisible: this._areCommunicationPrefsVisible()
       };
     },
 
     events: {
-      // validateAndSubmit is used to prevent multiple concurrent submissions.
-      'click #signout': BaseView.preventDefaultThen('validateAndSubmit')
+      'click #signout': BaseView.preventDefaultThen('signOut')
+    },
+
+    _onProfileChange: function () {
+      this._showAvatar();
+    },
+
+    // When we navigate to settings from a subview
+    // close the modal, show any ephemeral messages passed to `navigate`
+    _onNavigateFromSubview: function () {
+      if ($.modal.isActive()) {
+        $.modal.close();
+      }
+      this.showEphemeralMessages();
+      this.logScreen();
+    },
+
+    showSubView: function (SubView, options) {
+      var self = this;
+      if (self._panelViews.indexOf(SubView) === -1) {
+        return;
+      }
+
+      // Destroy any previous modal view
+      if (self._currentSubView && self._currentSubView.isModal) {
+        self._currentSubView.closePanel();
```

Comment:
Really, the same question for all "panel" management. I see `openPanel` is on the subview as well. I can imagine lots of reasons, just want to understand how logic was placed on the subview instead of the parent view.